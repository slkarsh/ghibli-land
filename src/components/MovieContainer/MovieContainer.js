import React from 'react'
import { connect } from 'react-redux'
import MovieCard from '../MovieCard/MovieCard'
import './MovieContainer.scss'

export const MovieContainer = ({ movies }) => {
  const display = movies.map((movie, index) => {
    return(
          <MovieCard
            {...movie}
            key={index}
          />
        )
  })

  return (
    <main className='movie-container'>
      {display}
    </main>
  )
}

 export const mapStateToProps = state => ({
  movies: state.movies
})

export default connect(mapStateToProps)(MovieContainer)