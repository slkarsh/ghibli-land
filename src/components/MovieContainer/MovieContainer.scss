.movie-container {
  display: grid;
  grid-gap: 80px 30px;
  padding: 15px 80px;
  grid-template-columns: repeat(auto-fit, minmax(330px, 2fr));
  overflow: scroll;
}