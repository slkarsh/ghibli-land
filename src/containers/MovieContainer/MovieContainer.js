import React from 'react'
import { connect } from 'react-redux'
import MovieCard from '../../components/MovieCard/MovieCard'
import './MovieContainer.scss'
import PropTypes from 'prop-types'

export const MovieContainer = ({ movies }) => {
  const display = movies.map((movie, index) => {
    return(
          <MovieCard
            {...movie}
            key={index}
          />
        )
  })

  return (
    <main className='movie-container'>
      {display}
    </main>
  )
}

 export const mapStateToProps = state => ({
  movies: state.movies
})

export default connect(mapStateToProps)(MovieContainer)

MovieContainer.propTypes = {
  movies: PropTypes.array
}