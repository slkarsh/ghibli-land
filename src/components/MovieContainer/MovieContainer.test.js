import { MovieContainer, mapStateToProps } from './MovieContainer';
import { shallow } from 'enzyme';
import React from 'react';




describe('MovieContainer', () => {
  describe('MovieContainer component', () => {
    let wrapper;
    beforeEach(() => {
      const mockMovies = [
        {
          title: 'A Film',
          description: 'it is riveting',
          release_date: '2019',
          rt_score: 22
        },
        {
          title: 'A Movie',
          description: 'it is not as riveting',
          release_date: '2018',
          rt_score: 11
        }
      ]
      wrapper = shallow(<MovieContainer
          movies={mockMovies}
         />)
    })

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    })
  })
})




//     it('should match a snapshot', () => {
//       expect(wrapper).toMatchSnapshot();
//     });

//     it('should render an error message', () => {
//       wrapper = shallow(<MovieContainer
//         movies={[{
//           title: 'Wicked',
//           release_date: '2017',
//           average_rating: 6.5,
//           popularity: 55.55,
//           id: 5
//         },
//         {
//           title: 'Joker',
//           release_date: '2019',
//           average_rating: 7.5,
//           popularity: 68.55,
//           id: 6
//         }]}
//         errorMsg='This is an error'
//         handleFavorite={mockHandleFavorite}
//       />)

//       expect(wrapper).toMatchSnapshot();
//     });
//   });

//   describe('mapStateToProps', () => {
//     it('should return an object with the todos array', () => {
//       // Setup
//       const mockState = {
//         movies: [{
//           title: 'Wicked',
//           release_date: '2017',
//           average_rating: 6.5,
//           popularity: 55.55,
//           movie_id: 5
//         },
//         {
//           title: 'Joker',
//           release_date: '2019',
//           average_rating: 7.5,
//           popularity: 68.55,
//           movie_id: 6
//         }],
//         user: {
//           name: 'Pol',
//           id: 2,
//           isSignedIn: true
//         },
//         errorMsg: ''

//       };
//       const expected = {
//         movies: [{
//           title: 'Wicked',
//           release_date: '2017',
//           average_rating: 6.5,
//           popularity: 55.55,
//           movie_id: 5
//         },
//         {
//           title: 'Joker',
//           release_date: '2019',
//           average_rating: 7.5,
//           popularity: 68.55,
//           movie_id: 6
//         }],
//         errorMsg: ''
//       };

//       // Execution
//       const mappedProps = mapStateToProps(mockState);

//       // Expectation
//       expect(mappedProps).toEqual(expected);
//     });
//   });

// })