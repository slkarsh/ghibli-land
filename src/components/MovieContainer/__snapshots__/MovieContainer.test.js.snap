// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieContainer MovieContainer component should match the snapshot 1`] = `
<main
  className="movie-container"
>
  <MovieCard
    description="it is riveting"
    key="0"
    release_date="2019"
    rt_score={22}
    title="A Film"
  />
  <MovieCard
    description="it is not as riveting"
    key="1"
    release_date="2018"
    rt_score={11}
    title="A Movie"
  />
</main>
`;
